gregBorchers/Homework2GregBorchers-cards-template1b.js

```diff
@@ -0,0 +1,263 @@
+                [ "Ace of Hearts"
+                , "Ace of Diamonds"
+                , "Ace of Spades"
+                , "Ace of Clubs"
+
+                , "Two of Hearts"
+                , "Two of Diamonds"
+                , "Two of Spades"
+                , "Two of Clubs"
+
+                , "Three of Hearts"
+                , "Three of Diamonds"
+                , "Three of Spades"
+                , "Three of Clubs"
+
+                , "Four of Hearts"
+                , "Four of Diamonds"
+                , "Four of Spades"
+                , "Four of Clubs"
+                                
+                , "Five of Hearts"
+                , "Five of Diamonds"
+                , "Five of Spades"
+                , "Five of Clubs"
+
+                , "Six of Hearts"
+                , "Six of Diamonds"
+                , "Six of Spades"
+                , "Six of Clubs"
+
+                , "Seven of Hearts"
+                , "Seven of Diamonds"
+                , "Seven of Spades"
+                , "Seven of Clubs"
+
+                , "Eight of Hearts"
+                , "Eight of Diamonds"
+                , "Eight of Spades"
+                , "Eight of Clubs"
+
+                , "Nine of Hearts"
+                , "Nine of Diamonds"
+                , "Nine of Spades"
+                , "Nine of Clubs"
+
+                , "Ten of Hearts"
+                , "Ten of Diamonds"
+                , "Ten of Spades"
+                , "Ten of Clubs"
+
+                , "Jack of Hearts"
+                , "Jack of Diamonds"
+                , "Jack of Spades"
+                , "Jack of Clubs"
+
+                , "Queen of Hearts"
+                , "Queen of Diamonds"
+                , "Queen of Spades"
+                , "Queen of Clubs"
+
+                , "King of Hearts"
+                , "King of Diamonds"
+                , "King of Spades"
+                , "King of Clubs" ];
+
+// **** alternative to build deckOfCards[] **** 
+// Loop across two string arrays to fill the array
+// Did not do this because having the static array was a useful visual reference
+// Here is code that would fill the array
+
+// var suitNames = ["Hearts", "Diamonds", "Spades", "Clubs"];
+// var rankNames = ["Ace","Two","Three","Four","Five","Six","Seven","Eight","Nine","Ten","Jack","Queen","King"];
+// var deckOfCards2 = [];
+// var cardID =0;
+
+// for (var i = 0; i < rankNames.length; i++) {
+//   for (var j = 0; j < suitNames.length; j++) {
+//     deckOfCards2[cardID] =  rankNames[i] + " of " + suitNames[j]
+//     console.log( deckOfCards2[cardID] );
+//     cardID++;
+//   };
+// };
+
+
+
+
+//****************************************** rank(card)
+// function rank(card) { // --> 1..13
+// }
+
+function rank(id) {
+  // validate inputs
+  if (!( (typeof id === 'number')
+      && (id%1 === 0) // is integer
+      && id < 52      // is in range
+      && id >= 0))    // is in range
```

Comment:
Just as elsewhere, I'll stress the value of dry code: rather than writing the error detection into each function, it's far better practice to have a helper function that you can call to see if a given input is valid. If the parameters for the input change at some point down the road (like say you want your program to work for Tarot decks, too), you'll have to update your code in a number of places. If you have a single validation function, you only have to change things in one place.